### DISCUSSION TITLE
关于同一个响应器如果有多个交互，应该如何用啥处理

### DISCUSSION THREAD

**Discussion by BigCat-byebye** [nonebot/discussions#430] [问答] [answered]
有个场景不太理解，比如有个填写订单的操作，用户使用create触发了这个响应后，需要用户填写很多项目，比如发货人，收货人，订单号这些，如何每次都用不同的promt的呢。。。got好像是并发的。。比如我写了个got 1 ，promt是1，got 2 是promt 2 ，但是这2个got好像一起执行了，并没有按照got 1执行，等待用户输入，got 2执行，等待用户输入，这种顺序。。。。啊。。。这种场景。。。应该怎么搞。。。

**Comment by BigCat-byebye** [comment]
如图，我需要先收集信息A，等待用户输入完成后，再收集B，再等得用户输入完成，应该是有序的

![image](https://github.com/nonebot/discussions/assets/32433624/15a302e2-e41a-4627-85a8-e0c8278f4976)


但是我测试的结果，确实2个got被同时执行了

![image](https://github.com/nonebot/discussions/assets/32433624/19e2f0df-10ed-4141-8a62-d38c55c28f1a)


这是我姿势不对么，还是啥。。。

**Reply by yanyongyu** [reply]
你为什么每个got前面都写个on_command？

**Reply by BigCat-byebye** [reply]
因为这是同一个事件。。。。都是create reply rule

**Reply by BigCat-byebye** [reply]
我看官网的weather不也是on command 么，后面的handler和got也是weather的装饰器啊

**Reply by BigCat-byebye** [reply]
。。。。好的，我找到了。好像不能这么干。。。。应该定义1个test1=on_command，然后再test.handle和test.got，这样就会按照顺序了。。

**Comment by BigCat-byebye** [comment, marked as answer]
应该定义1个test1=on_command，然后再test.handle和test.got，这样就会按照顺序
